From ccf7e2520a1c75a035de140ceef92d5a822371ef Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:13:03 -0400
Subject: [powerpc] powernv: Pass buffer size to OPAL validate flash call

Message-id: <1410545655-205645-555-git-send-email-dzickus@redhat.com>
Patchwork-id: 94451
O-Subject: [RHEL7 PATCH 554/626] powerpc/powernv: Pass buffer size to OPAL validate flash call
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit 8b8f7bf4c218628fd243d03fc85cdbc7039e9e35
Author: Vasant Hegde <hegdevasant@linux.vnet.ibm.com>
Date:   Thu Jun 5 10:13:40 2014 +0530

    powerpc/powernv: Pass buffer size to OPAL validate flash call

    We pass actual buffer size to opal_validate_flash() OPAL API call
    and in return it contains output buffer size.

    Commit cc146d1d (Fix little endian issues) missed to set the size
    param before making OPAL call. So firmware image validation fails.

    This patch sets size variable before making OPAL call.

    Signed-off-by: Vasant Hegde <hegdevasant@linux.vnet.ibm.com>
    Tested-by: Thomas Falcon <tlfalcon@linux.vnet.ibm.com>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/platforms/powernv/opal-flash.c b/arch/powerpc/platforms/powernv/opal-flash.c
index 1a98617..f7faf65 100644
--- a/arch/powerpc/platforms/powernv/opal-flash.c
+++ b/arch/powerpc/platforms/powernv/opal-flash.c
@@ -131,7 +131,8 @@ static inline void opal_flash_validate(void)
 {
  long ret;
  void *buf = validate_flash_data.buf;
- __be32 size, result;
+ __be32 size = cpu_to_be32(validate_flash_data.buf_size);
+ __be32 result;
 
  ret = opal_validate_flash(__pa(buf), &size, &result);
 
-- 
1.7.1